### PR TITLE
fix(cli): dispatch reason run and store connect to real APIs

### LINKED ISSUE #1354: [BUG] semantica reason run fails with 'Reasoner' object has no attribute 'run' (and store connect has a sibling dispatch error)
## Summary

`semantica reason run` fails immediately:

```
'Reasoner' object has no attribute 'run'
```

The CLI dispatches to a method that does not exist on `Reasoner`.

## Reproduction

```bash
export GRAPH_STORE_DEFAULT_BACKEND=neo4j
export GRAPH_STORE_NEO4J_URI=bolt://127.0.0.1:7689
export GRAPH_STORE_NEO4J_USER=neo4j
export GRAPH_STORE_NEO4J_PASSWORD=<redacted>

semantica reason run
# -> Semantica error: 'Reasoner' object has no attribute 'run'
```

`semantica reason --help` lists `run` as a documented subcommand ("Execute a reasoning engine against the knowledge graph").

## Analysis

`Reasoner` exposes `infer_facts` (the v0.6.7 changelog notes a `reasoning_provenance.py` import fix that renamed `ReasoningEngine`/`infer` to `Reasoner`/`infer_facts`). The CLI's `reason run` path appears to have been missed by that rename.

## Related

`semantica store connect --backend neo4j` also fails on a dispatch error in the same family:

```
get_graph_store_method() missing 1 required positional argument: 'method_name'
```

Notably it fails **identically** with correct credentials, deliberately wrong credentials, and no configuration at all — so it never reaches a connection attempt, and cannot be used to diagnose connectivity. Happy to split that into its own issue if preferred.

## Environment

semantica 0.6.7 (tag), Python 3.12.14, macOS arm64, Neo4j 2026.07.1 Community, neo4j driver 6.2.0.

### PR DESCRIPTION
**Pull request by 7487**
## Description

Two CLI dispatch failures reported in #1354, both raised before doing any real work:

- `semantica reason run` called `Reasoner.run()`, which does not exist — the facade's API is `infer_facts(facts, rules)`. The command now reads nodes/relationships from the configured graph store and converts them to fact strings using the same conventions `Reasoner.add_fact()` applies to KG-style dicts (`Label(name)`, `TYPE(source, target)` with internal ids resolved to names), loads `--rules` as a YAML list/`rules:` mapping or plain-text lines, and reports `engine` / `facts` / `inferred_count` / `inferred_facts`.
- `semantica store connect` called `get_graph_store_method(backend)`, which is the `(task, method_name)` method registry, not a backend factory — so it raised `TypeError: missing 1 required positional argument: 'method_name'` before any connection attempt, failing identically with correct credentials, wrong credentials, or none. It now builds the store via `GraphStore(backend=...)` and probes `connect()`, so real connectivity/auth errors surface (`Neo4j authentication failed`, `service unavailable`, `Unknown backend: ...`).

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

## Related Issues

Fixes #1354

## Changes Made

- `semantica/cli.py`: `reason_run` dispatches to `Reasoner.infer_facts()`; new helpers `_load_rule_definitions()` and `_graph_store_facts()` (reusing the existing `_get_graph_store()`); `store_connect` builds the backend through `GraphStore` and keeps the existing probe loop.
- `tests/test_cli_commands.py`: regression tests — `reason run` end-to-end with a fake store and the real `Reasoner` (`--rules` YAML → inferred facts in `--json` output), rules-file format handling, and `store connect` dispatch through `GraphStore` including `--uri` override.

## Testing

- [x] Tested locally
- [x] Added tests for new functionality

`pytest tests/test_cli_commands.py -k "TestReason or TestStore"` → 26 passed, 1 pre-existing failure (`test_migrate_refuses_unsupported_backend_pair`, terminal-width panel wrapping, fails identically on clean `main`). Full `tests/test_cli_commands.py` + `tests/test_cli_foundation.py`: same pass/fail set as clean `main` plus the 3 new tests passing.

Manual before/after:

```
$ semantica reason run
Semantica error: 'Reasoner' object has no attribute 'run'   # before
Failed to get nodes: Neo4j is not available ...              # after (real store error)

$ semantica store connect --backend neo4j
Error: get_graph_store_method() missing 1 required positional argument: 'method_name'  # before
Error: Neo4j is not available. Install it with: pip install neo4j                      # after
```

## Documentation

- [x] No documentation changes needed

## Breaking Changes

**Breaking Changes**: No

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

## Additional Notes

Commented on #1354 to claim it per the CONTRIBUTING workflow; opening this for triage and happy to adjust per the assignment rules. The `reason run` semantics implemented here are the minimal honest dispatch — datalog/SPARQL rule files beyond IF/THEN strings still route through their own engines (`reason query`).

### CODE REVIEW

File: semantica/cli.py [hunk 1]
```diff
@@ -1155,6 +1155,51 @@ def _get_graph_store(cli_ctx: CLIContext) -> Any:
...
+    for node in nodes:
+        props = node.get("properties") or {}
+        name = props.get("name") or props.get("id") or node.get("id")
+        names[node.get("id")] = name
+        label = (node.get("labels") or ["Entity"])[0]
+        facts.append(f"{label}({name})")
+    for rel in relationships:
+        source = names.get(rel.get("start_id"), rel.get("start_id"))
+        target = names.get(rel.get("end_id"), rel.get("end_id"))
+        facts.append(f"{rel.get('type', 'RELATED_TO')}({source}, {target})")
```
[the remark this answers is not in the document]
Comment (7487) [reply semantica/cli.py]: Confirmed — all four backends (neo4j, falkordb, neptune, age) return `start_node_id`/`end_node_id` from `get_relationships()`, so the helper's `start_id`/`end_id` lookups always produced `TYPE(None, None)`. Fixed in fe3490ce: the helper now reads the real keys, and the regression test's fake store uses the backend schema plus a `MANAGES(?x, ?y)` rule that only passes when endpoints resolve to names.

File: semantica/cli.py [hunk 2]
```diff
@@ -2162,15 +2207,30 @@ def reason_run(cli_ctx: CLIContext, engine: str, rules: Optional[str],
...
             from .reasoning import Reasoner
+            # Reasoner has no run() method (#1354); dispatch to its real
+            # API: facts from the configured graph store + rules from the
+            # optional --rules file into infer_facts().
             r = Reasoner(engine=engine, config=cli_ctx.config.to_dict())
+            rule_defs = _load_rule_definitions(rules) if rules else None
+            facts = _graph_store_facts(cli_ctx)
+
+            def _infer() -> Dict[str, Any]:
+                inferred = r.infer_facts(facts, rule_defs)
```
[the remark this answers is not in the document]
Comment (7487) [reply semantica/cli.py:2229]: True — `Reasoner` stores `engine` in config without reading it, and `infer_facts()` always forward-chains. Rather than growing this dispatch fix into per-engine wiring (the dedicated engines take different inputs entirely: SPARQL/Datalog queries, observations for abductive, `Premise` objects for deductive — and `DatalogReasoner` rejects capitalized constants like real node names), fe3490ce makes the command honest: `rete`/`forward-chain` run through `infer_facts()`, and the other five engine values now fail up front with a clear message pointing SPARQL/Datalog users at `reason query` instead of reporting an engine that never ran. Before this PR all seven choices crashed on `Reasoner.run`, so no working behavior changes. Wiring the dedicated engines into `reason run` is a feature I'd keep as follow-up scope.

File: semantica/cli.py [hunk 3]
```diff
@@ -1155,6 +1155,51 @@ def _get_graph_store(cli_ctx: CLIContext) -> Any:
...
+    nodes = gs.get_nodes(limit=sys.maxsize)
+    relationships = gs.get_relationships(limit=sys.maxsize)
+    names: Dict[Any, Any] = {}
+    facts: List[str] = []
+    for node in nodes:
+        props = node.get("properties") or {}
+        name = props.get("name") or props.get("id") or node.get("id")
+        names[node.get("id")] = name
+        label = (node.get("labels") or ["Entity"])[0]
+        facts.append(f"{label}({name})")
```
[the remark this answers is not in the document]
Comment (7487) [reply semantica/cli.py]: Fixed in fe3490ce — one fact per label instead of `labels[0]`, with a multi-label node (`Person`+`Employee`) and a rule on the secondary label in the regression test.

**Review by Sameer6305 — APPROVED** [review]
## Approved ✅

Reviewed PR #1372 against issue #1354.

Flow:

* `reason run` was calling the removed `Reasoner.run()` API → fixed to use `Reasoner.infer_facts()`.
* Graph data is converted into the fact format expected by the reasoning engine.
* `store connect` was incorrectly using `get_graph_store_method()` → fixed to construct the store through `GraphStore` and perform a real connection probe.
* Added regression coverage for reasoning, rule loading, store dispatch, errors, and JSON output.
* During review, one edge case in YAML rule loading was found and fixed locally.
* Relevant tests pass; remaining failures were verified as pre-existing on `main`.

**Approved — changes look correct and address the reported issue.**
